GUIDELINES.md

```diff
@@ -207,6 +207,46 @@ Make sure all tests are passing with:
 
     $ cargo test --all-features
 
+### Running end-to-end tests
+
+In order to run end-to-end (e2e) tests you need to have a specific nightly toolchain.
+"Nightly" is necessary to use optimization compiler flags and have contract wasm small enough to be eligible for
+deployment.
+
+Run the following commands to install the necessary toolchain:
+
+```shell
+rustup install nightly-2024-01-01
+rustup component add rust-src
+```
+
+Also, you should have the cargo stylus tool:
+
+```shell
+cargo install cargo-stylus
+```
+
+Since most of the e2e tests use [koba](https://github.com/OpenZeppelin/koba) for deploying contracts, you need to
+install the solidity compiler (`v0.8.21`). You can do so, following these steps:
+this:
+
+```shell
+npm install -g solc@0.8.21
```

Comment:
IMHO missing information how to install `npm`

Reply:
True.... but I don't think we should teach people how to install every tool as part of this guide. I'm happy with just specifying it is a prerequisite. But @qalisander can decide what works best.

Reply:
It was an idea of @ggonzalez94 to use npm. I would prefer to have a straightforward installation guideline. And if user want to use `npm`, `yarn`, `brew` or `apt-get` he will decide. Wdt guys?

Reply:
I'm in favor of npm, I just think that people can figure out how to install it themselves(or we can put a link to the official docs). But I'm happy just listing it a prerequisite :)

Reply:
`npm` is fine for me, I think that we need to mention that it is a prerequisite and link to the official docs.

Reply:
IMHO the best way is just to point that we require this specific version and leave link to this - https://docs.soliditylang.org/en/latest/installing-solidity.html#